clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -355,6 +355,10 @@ public void close() {
     }
 
     public void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+        commitOffsetsAsync(offsets, callback, true);
+    }
+
+    private void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback, boolean executeDelayedTasks) {
```

Comment:
You might want to add a few notes about why the `executeDelayedTasks` even exists. It's very non-obvious.